### PR TITLE
[sui-proxy/ cleanup handler/middleware]

### PR DESCRIPTION
**Pull request by suiwombat**
Summary:

* cleanup the consumer.rs code
* add a new middleware to decode the protobufs
* improve error handling for rpc call to detect general rpc issues

Test Plan:

tested locally

## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-proxy/src/consumer.rs [hunk 1]
```diff
@@ -52,64 +53,109 @@ impl ProtobufDecoder {
...
             m.mut_label().extend(labels.clone());
         }
     }
+    data
+}
+
+fn encode_compress(request: &WriteRequest) -> Result<Vec<u8>, (StatusCode, &'static str)> {
+    let mut buf = Vec::new();
+    buf.reserve(request.encoded_len());
+    let Ok(()) = request.encode(&mut buf) else {
```
Comment (after-ephemera) [inline crates/sui-proxy/src/consumer.rs:83]: nit
```suggestion
    if request.encode(&mut buf).is_err() {
```

**Review by after-ephemera — APPROVED** [review]
lgtm

File: crates/sui-proxy/src/consumer.rs [hunk 2]
```diff
@@ -52,64 +53,109 @@ impl ProtobufDecoder {
...
+            Ok(())
+        }
+        reqwest::StatusCode::BAD_REQUEST => {
+            error!("TRIED: {:?}", request);
+            let body = response
+                .text()
+                .await
+                .unwrap_or_else(|_| "response body cannot be decoded".into());
+
+            if body.contains("err-mimir-sample-out-of-order") {
```
[the remark this answers is not in the document]
Comment (suiwombat) [reply crates/sui-proxy/src/consumer.rs:121]: unfortunately, the error comes from mimir and it doesn't explain why, it just says it.  It's an error we don't have any control over and it's also not actionable.  I'm not even sure what i'd write here as a comment.

File: crates/sui-proxy/src/consumer.rs (hunk 1 above)
Comment (suiwombat) [reply crates/sui-proxy/src/consumer.rs:83]: i'll fix this on the next pr. thanks <3

**Pull request merged**